### PR TITLE
Remove unnecessary copies of ParquetCachedBatchSerializer

### LINKED ISSUE #2905: Remove Spark version specific Serializers unless they need to override behavior
Right now we have ParquetCachedBatchSerializer for every version of Spark since 3.1.1 e.g 312 311db and 311cdh. We need to come up with a way to avoid this unnecessary class.

### PR DESCRIPTION
**Pull request by razajafri**
We decided to remove the extra serializers and have a single version of it in the 311 shims. The package name will be changed to better suit whatever strategy we come up as a part of supporting Spark 3.2 

Fixes #2905 

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

### CODE REVIEW

**Comment by razajafri** [comment]
build

**Review by jlowe — COMMENTED** [review]
Looks like a spark320 version was missed?

**Comment by razajafri** [comment]
> Looks like a spark320 version was missed?

Sorry, I left it out for the draft, but forgot to remove it post our meeting

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by jlowe — CHANGES_REQUESTED** [review]
Actually the documentation does need to be changed here.  The cache docs refer to a table showing different classes for each Spark version, and those are being deleted in this PR.

**Review by jlowe — APPROVED** [review]

**Comment by razajafri** [comment]
build

**Review by jlowe — CHANGES_REQUESTED** [review]
It looks like jenkins/spark-tests.sh also needs to be updated since it tries to pick the shim-specific class.

**Comment by razajafri** [comment]
build

File: jenkins/spark-tests.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ spark-submit $BASE_SPARK_SUBMIT_ARGS $CUDF_UDF_TEST_ARGS --jars $RAPIDS_TEST_JAR
 #only run cache tests with our serializer in nightly test for Spark version >= 3.1.1
 if [[ "$IS_SPARK_311_OR_LATER" -eq "1" ]]; then
   SHIM_PACKAGE=$(echo ${SPARK_VER} | sed 's/\.//g' | sed 's/-SNAPSHOT//')
```
Comment (jlowe) [inline jenkins/spark-tests.sh:106]: This is no longer needed.
```suggestion
```

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**